File: rb/features.rb
# Postcodesio SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module PostcodesioFeatures
  def self.make_feature(name)
    case name
    when "base"
      PostcodesioBaseFeature.new
    when "test"
      PostcodesioTestFeature.new
    else
      PostcodesioBaseFeature.new
    end
  end
end

File: py/features.py
# Postcodesio SDK feature factory

from feature.base_feature import PostcodesioBaseFeature
from feature.test_feature import PostcodesioTestFeature


def _make_feature(name):
    features = {
        "base": lambda: PostcodesioBaseFeature(),
        "test": lambda: PostcodesioTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
